### PR TITLE
Add spec file for building tendrl-alerting

### PR DESCRIPTION
**Pull request by TimothyAsirJeyasing**
This path does the following things:
1) Update MANIFEST.in with the required file details
2) Update setup.py to bump version
3) Add Makefile and spec file for build this package
4) Add service file to start the tendrl-alerting service

tendrl-bug-id: Tendrl/alerting#27
Signed-off-by: Timothy Asir J <tjeyasin@redhat.com>

### CODE REVIEW

**Comment by TimothyAsirJeyasing** [comment]
@r0h4n , @nthomas-redhat , @shtripat , @anmolbabu , Please review

File: tendrl-alerting.spec [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+# remove the sphinx-build leftovers
+rm -rf html/.{doctrees,buildinfo}
+
+%install
+%{__python} setup.py install --single-version-externally-managed -O1 --root=$RPM_BUILD_ROOT --record=INSTALLED_FILES
+install -m  0755  --directory $RPM_BUILD_ROOT%{_var}/log/tendrl/alerting
+install -m  0755  --directory $RPM_BUILD_ROOT%{_sysconfdir}/tendrl/alerting
+install -m  0755  --directory $RPM_BUILD_ROOT%{_datadir}/tendrl/alerting
+install -Dm 0644 tendrl-alerting.service $RPM_BUILD_ROOT%{_unitdir}/tendrl-alerting.service
+install -Dm 0644 etc/tendrl/tendrl.conf.sample $RPM_BUILD_ROOT%{_sysconfdir}/tendrl/tendrl.conf
```
Comment (anmolbabu) [inline tendrl-alerting.spec]: It is not tendrl.conf now

File: tendrl-alerting.spec [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+
+BuildRequires: python-gevent
+BuildRequires: pytest
+BuildRequires: systemd
+BuildRequires: python-mock
+
+Requires: python-etcd
+Requires: python-dateutil
+Requires: gevent
+Requires: python-greenlet
```
Comment (anmolbabu) [inline tendrl-alerting.spec]: python-greenlet, gevent Not required

File: tendrl-alerting.spec [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+Requires: msgpack-python
+Requires: pytz
+Requires: flask
+Requires: Flask-API
```
Comment (anmolbabu) [inline tendrl-alerting.spec]: Requires: tendrl-commons
Requires: six
Requires: PyYAML

File: tendrl-alerting.spec (hunk 1 above)
Comment (shtripat) [inline tendrl-alerting.spec]: Also the source file path should be etc/tendrl/alerting/alerting.conf.yaml.sample

File: tendrl-alerting.spec [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+rm -rf html/.{doctrees,buildinfo}
+
+%install
+%{__python} setup.py install --single-version-externally-managed -O1 --root=$RPM_BUILD_ROOT --record=INSTALLED_FILES
+install -m  0755  --directory $RPM_BUILD_ROOT%{_var}/log/tendrl/alerting
+install -m  0755  --directory $RPM_BUILD_ROOT%{_sysconfdir}/tendrl/alerting
+install -m  0755  --directory $RPM_BUILD_ROOT%{_datadir}/tendrl/alerting
+install -Dm 0644 tendrl-alerting.service $RPM_BUILD_ROOT%{_unitdir}/tendrl-alerting.service
+install -Dm 0644 etc/tendrl/tendrl.conf.sample $RPM_BUILD_ROOT%{_sysconfdir}/tendrl/tendrl.conf
+install -Dm 0644 etc/logging.yaml.timedrotation.sample $RPM_BUILD_ROOT%{_sysconfdir}/tendrl/alerting_logging.yaml
```
Comment (shtripat) [inline tendrl-alerting.spec]: Same above mentioned location for logging configuration as well.

**Comment by r0h4n** [comment]
@TimothyAsir why is this still pending?

File: tendrl-alerting.service [hunk 5]
```diff
@@ -0,0 +1,13 @@
+[Unit]
```
Comment (anmolbabu) [inline tendrl-alerting.service]: Please keep this inline with https://github.com/Tendrl/performance-monitoring/blob/master/tendrl-performance-monitoring.service

File: tendrl-alerting.spec (hunk 1 above)
Comment (anmolbabu) [inline tendrl-alerting.spec]: Posting my local diff here:

-install -Dm 0644 etc/tendrl/tendrl.conf.sample $RPM_BUILD_ROOT%{_sysconfdir}/tendrl/tendrl.conf
-install -Dm 0644 etc/logging.yaml.timedrotation.sample $RPM_BUILD_ROOT%{_sysconfdir}/tendrl/alerting_logging.yaml
+install -Dm 0644 etc/tendrl/alerting.conf.yaml $RPM_BUILD_ROOT%{_sysconfdir}/tendrl/alerting/alerting.conf.yaml
+install -Dm 0644 etc/tendrl/alerting_logging.yaml $RPM_BUILD_ROOT%{_sysconfdir}/tendrl/alerting/alerting_logging.yaml
+install -Dm 0644 etc/tendrl/email.conf.yaml $RPM_BUILD_ROOT%{_sysconfdir}/tendrl/alerting/email.conf.yaml


-%{_sysconfdir}/tendrl/tendrl.conf
-%{_sysconfdir}/tendrl/alerting_logging.yaml
+%{_sysconfdir}/tendrl/alerting/alerting.conf.yaml
+%{_sysconfdir}/tendrl/alerting/alerting_logging.yaml
+%{_sysconfdir}/tendrl/alerting/email.conf.yaml

Comment (TimothyAsirJeyasing) [inline tendrl-alerting.spec]: Done

File: tendrl-alerting.spec (hunk 2 above)
Comment (TimothyAsirJeyasing) [inline tendrl-alerting.spec]: Done

File: tendrl-alerting.spec (hunk 3 above)
Comment (TimothyAsirJeyasing) [inline tendrl-alerting.spec]: you want a package called six or python-six. both packages are available

File: tendrl-alerting.spec (hunk 4 above)
Comment (TimothyAsirJeyasing) [inline tendrl-alerting.spec]: Done

File: tendrl-alerting.spec (hunk 3 above)
Comment (anmolbabu) [inline tendrl-alerting.spec]: I need python-six

**Comment by anmolbabu** [comment]
@TimothyAsirJeyasing Please rebase this patch looks like there'll be no requirements.txt now and the dependencies are moved to setup.py itself in latest master as per https://github.com/Tendrl/alerting/pull/38.

**Comment by anmolbabu** [comment]
Rest LGTM

**Comment by r0h4n** [comment]
@TimothyAsirJeyasing Please update this, its long pending

**Comment by anmolbabu** [comment]
@r0h4n @TimothyAsirJeyasing I have taken this commit with tim's PR signature maintained into https://github.com/Tendrl/alerting/pull/41 So I think this can be abandoned

**Comment by TimothyAsirJeyasing** [comment]
Closing this because its already merged along with #41

**Pull request closed without merging**